### ISSUE TITLE
Add missing security token support

### ISSUE THREAD

**Issue by confluence** [idia-astro/carta-python#13]
The wrapper needs to use the security token when opening a new frontend session. We should probably also restrict access to the gRPC interface using the same token. We need to decide on and implement these security features before we can move forward with merging of the development branch and wider publication of this module (on PyPi and Read The Docs).

Adding authentication to the gRPC interface would require a change to the backend as well.

**Comment by confluence** [comment]
I am closing this issue and will open a new issue for SDM support.
